pkg/preparer/orchestrate.go

```diff
@@ -226,6 +227,12 @@ func (p *Preparer) installAndLaunchPod(newManifest *pods.PodManifest, pod Pod, l
 			logger.WithField("err", err).Errorln("Service registration failed")
 			return false
 		}
+
+		if currentManifest != nil {
+			// the exit code of disable is always ignored
+			pod.Disable(currentManifest)
```

Comment:
On the topic of ordering and formal constraints on what we're doing, lets take a look at how we can improve orchestrate.go. We have a task to establish the failure cases concretely, it might help to structure this file better and write some automation to enforce our expectations a bit more tightly.